plugins/basic/nexus-repository-maven/src/main/java/org/sonatype/nexus/repository/maven/internal/MavenFacetImpl.java

```diff
@@ -112,22 +113,39 @@
 
   /**
    * List of hash algorithms always applied to payloads in blob store, with first in the list being the "primary" hash.
+   * In case of Maven, this list is driven by {@link HashType}.
+   *
+   * @see HashType
    */
-  private static final List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.of(SHA1, MD5);
+  private static final List<HashAlgorithm> HASH_ALGORITHMS = ImmutableList.copyOf(
```

Comment:
Heh, maybe just a constant in HashType ?
    public final static List<HashAlgorithm> ALGORITHMS = ImmutableList
        .of(SHA1.getHashAlgorithm(), MD5.getHashAlgorithm());